# HG changeset patch
# Parent ac7fe0216dfa365ff3de4374a46eed834e8f488f
# User Ben Kelly <ben@wanderview.com>
Make sure the Context QuotaInfo is set before handling errors in OnQuotaInit().


diff --git a/dom/cache/Context.cpp b/dom/cache/Context.cpp
--- a/dom/cache/Context.cpp
+++ b/dom/cache/Context.cpp
@@ -536,26 +536,27 @@ Context::DispatchAction(nsIEventTarget* 
 }
 
 void
 Context::OnQuotaInit(nsresult aRv, const QuotaInfo& aQuotaInfo)
 {
   NS_ASSERT_OWNINGTHREAD(Context);
   MOZ_ASSERT(mState == STATE_CONTEXT_INIT);
 
+  mQuotaInfo = aQuotaInfo;
+
   if (NS_FAILED(aRv)) {
     for (uint32_t i = 0; i < mPendingActions.Length(); ++i) {
       mPendingActions[i].mAction->CompleteOnInitiatingThread(aRv);
     }
     mPendingActions.Clear();
     // Context will destruct after return here and last ref is released.
     return;
   }
 
-  mQuotaInfo = aQuotaInfo;
   mState = STATE_CONTEXT_READY;
 
   for (uint32_t i = 0; i < mPendingActions.Length(); ++i) {
     DispatchAction(mPendingActions[i].mTarget, mPendingActions[i].mAction);
   }
   mPendingActions.Clear();
 }
 
